TestLib/XCTestCase/XCUIApplication+GREYEnvironment.m

```diff
@@ -23,7 +23,7 @@ - (void)grey_configureApplicationForLaunch {
   NSString *insertionKey = @"DYLD_INSERT_LIBRARIES";
   NSString *insertionValue = @"@executable_path/Frameworks/AppFramework.framework/AppFramework";
   NSString *alreadyExistingValue = [mutableEnv valueForKey:insertionKey];
-  if (alreadyExistingValue) {
+  if (alreadyExistingValue && ![alreadyExistingValue containsString:insertionValue]) {
```

Comment:
The substring check isn't comprehensive enough. What you should do instead if check if any of the individual values contains the AppFramework check.

```
  NSArray *existingValues = [alreadyExistingValue componentsSeparatedByString:@":"];
  if (![[NSSet setWithArray:existingValues] containsObject:insertionValue]) {
    insertionValue = [NSString stringWithFormat:@"%@:%@", alreadyExistingValue, insertionValue];
  }
```

You could then remove the if check completely in line 26 with this.

Reply:
I have introduced the Set based check as suggested. However, if check can not be completely removed. With `:` instead of `,` used as argument separator, even duplicate arguments are being accepted by iOS. However, an if check for `existingValues` is still required to not append the new argument to `nil` value.